ehsm_kms_service/secret_manager_apis.js

```diff
@@ -234,9 +234,68 @@ const createSecret = async (res, appid, payload, DB) => {
     }
 }
 
+//update the description of secret
+const updateSecretDesc = async (res, appid, payload, DB) => {
+    //get and check param in payload
+    let secretName = getParam_String(payload, 'secretName')
+    let description = getParam_String(payload, 'description')
+    if (!checkStringParam(secretName, true)) {
+        res.send(_result(400, 'secretName cannot be empty and must be string'))
+        return
+    }
+    if (!checkStringParam(description, true)) {
+        res.send(_result(400, 'description cannot be empty and must be string'))
+        return
+    }
+
+    //Query the description through secret name and update the description
+    const secret_name_query = {
+        selector: {
+            appid,
+            secretName
+        },
+        fields: [
+            '_id',
+            '_rev',
+            'appid',
+            'secretName',
+            'encryptionKeyId',
+            'description',
+            'createTime',
+            'deleteTime',
+            'plannedDeleteTime',
+            'rotationInterval',
+            'lastRotationDate',
+            'nextRotationDate'
+        ],
+        limit: 1,
+    }
+    await DB.partitionedFind('secret_metadata', secret_name_query)
+        .then(async (secret_metadata_res) => {
+            if (secret_metadata_res.docs.length > 0) {
```

Comment:
If I want to update my description from "des" to "". Will this code block this?

Reply:
""is not supported，it will sign error

Reply:
payload["description"] = ""
why sign error?

Reply:
yes

Reply:
”“ in any payload will cause signerror

Reply:
That should be a sign problem. We should fix this and support "". Will discuss it later.

Reply:
Got it